plugins/inputs/ravendb/ravendb.go

```diff
@@ -0,0 +1,430 @@
+)
+
+// defaultURL will set a default value that corresponds to the default value
+// used by RavenDB
+const defaultURL = "http://localhost:8080"
+
+const defaultTimeout = 5
+
+// RavenDB defines the configuration necessary for gathering metrics,
+// see the sample config for further details
+type RavenDB struct {
+	URL  string `toml:"url"`
+	Name string `toml:"name"`
+
+	Timeout internal.Duration `toml:"timeout"`
+
+	StatsInclude       []string `toml:"stats_include"`
+	DbStatsDbs         []string `toml:"db_stats_dbs"`
+	IndexStatsDbs      []string `toml:"index_stats_dbs"`
+	CollectionStatsDbs []string `toml:"collection_stats_dbs"`
+
+	tls.ClientConfig
+
+	Log telegraf.Logger `toml:"-"`
+
+	client               *http.Client
+	requestUrlServer     string
+	requestUrlDatabases  string
+	requestUrlIndexes    string
+	requestUrlCollection string
+}
+
+var sampleConfig = `
+  ## Node URL and port that RavenDB is listening on.
+  url = "https://localhost:8080"
+
+  ## RavenDB X509 client certificate setup
+  # tls_cert = "/etc/telegraf/raven.crt"
+  # tls_key = "/etc/telegraf/raven.key"
+
+  ## Optional request timeout
+  ##
+  ## Timeout, specifies the amount of time to wait
+  ## for a server's response headers after fully writing the request and 
+  ## time limit for requests made by this client.
+  # timeout = "5s"
+
+  ## List of statistics which are collected
+  # At least one is required
+  # Allowed values: server, databases, indexes, collections
+  #
+  # stats_include = ["server", "databases", "indexes", "collections"]
+
+  ## List of db where database stats are collected
+  ## If empty, all db are concerned
+  # db_stats_dbs = []
+
+  ## List of db where index status are collected
+  ## If empty, all indexes from all db are concerned
+  # index_stats_dbs = []
+
+  ## List of db where collection status are collected
+  ## If empty, all collections from all db are concerned
+  # collection_stats_dbs = []
+`
+
+// SampleConfig ...
+func (r *RavenDB) SampleConfig() string {
+	return sampleConfig
+}
+
+// Description ...
+func (r *RavenDB) Description() string {
+	return "Reads metrics from RavenDB servers via the Monitoring Endpoints"
+}
+
+// Gather ...
+func (r *RavenDB) Gather(acc telegraf.Accumulator) error {
+	err := r.ensureClient()
+	if nil != err {
+		r.Log.Errorf("Error with Client %s", err)
+		return err
+	}
+
+	var wg sync.WaitGroup
+
+	for _, statToCollect := range r.StatsInclude {
+		switch statToCollect {
+		case "server":
+			wg.Add(1)
+
+			go func() {
+				defer wg.Done()
+				r.gatherServer(acc)
+			}()
+		case "databases":
+			wg.Add(1)
+
+			go func() {
+				defer wg.Done()
+				r.gatherDatabases(acc)
+			}()
+		case "indexes":
+			wg.Add(1)
+
+			go func() {
+				defer wg.Done()
+				r.gatherDatabases(acc)
+			}()
+		case "collections":
+			wg.Add(1)
+
+			go func() {
+				defer wg.Done()
+				r.gatherCollections(acc)
+			}()
+		default:
+			acc.AddError(fmt.Errorf("unhandled statistics type %s", statToCollect))
```

Comment:
Please check this in `Init()`. You can use internal/choice with `choice.CheckSlice(r.StatsInclude, []string{"server", "databases", "indexes", "collections"})` and error out early in `Init()`. This why we avoid spamming the logfile in case of an invalid choice.

Reply:
done - moved check to init